src/briefcase/integrations/virtual_environment.py

```diff
@@ -238,7 +268,8 @@ def virtual_environment(
     :returns: A context manager for an environment where code can be executed.
     """
     if not isolated:
-        return NoOpEnvironment(tools=tools, console=console)
+        marker_path = venv_path / ".venv-marker"
```

Comment:
Making this a hidden file will make it a little harder to visually inspect based on a directory listing.

I'd suggest that it might make more sense to make `venv_path` a *file* for the non-isolated case. That makes it clear that it's not an isolated virtual environment; and provides a basis to check for the "was/wasn't previously run in a non-isolated environment" in `VenvContext.exists()`/`check_and_update_marker()` (and prompt the user to see if they want to destroy the old configuration).